traffic_router/core/src/main/java/org/apache/traffic_control/traffic_router/core/loc/AbstractServiceUpdater.java

```diff
@@ -372,12 +373,14 @@ protected File downloadDatabase(final String url, final File existingDb) throws
 			if (conn instanceof HttpURLConnection) {
 				eTag = conn.getHeaderField("ETag");
 				if (((HttpURLConnection) conn).getResponseCode() == HttpURLConnection.HTTP_NOT_MODIFIED) {
-					LOGGER.info("[" + getClass().getSimpleName() + "] " + url + " not modified since our existing database's last update time of " + new Date(existingDb.lastModified()));
+					LOGGER.info("[" + getClass().getSimpleName() + "] " + url + " not modified since our existing database's last update time of " + new Date(existingLastModified));
 					return existingDb;
 				}
+			} else if (dbURL.getProtocol().equals("file") && conn.getLastModified() <= existingLastModified) {
```

Comment:
Should this condition should only succeed if the *last modified* time is greater than 0?

Reply:
Yeah, that should be fine as long as nobody makes any closed time-like loops